tests/integration_tests/modules/test_combined.py

```diff
@@ -424,6 +424,50 @@ def test_instance_json_gce(self, class_client: IntegrationInstance):
         assert v1_data["instance_id"] == client.instance.instance_id
         assert v1_data["local_hostname"] == client.instance.name
 
+    @pytest.mark.lxd_container
+    def test_instance_cloud_id_across_reboot(
+        self, class_client: IntegrationInstance
+    ):
+        client = class_client
+        assert client.execute("test -f /run/cloud-init/cloud-id-lxd").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+        client.restart()
+        assert client.execute("test -f /run/cloud-init/cloud-id-lxd").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+
+    @pytest.mark.azure
+    def test_instance_cloud_id_across_reboot_az(
+        self, class_client: IntegrationInstance
+    ):
+        client = class_client
+        assert client.execute("test -f /run/cloud-init/cloud-id-azure").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+        client.restart()
+        assert client.execute("test -f /run/cloud-init/cloud-id-azure").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+
+    @pytest.mark.gce
+    def test_instance_cloud_id_across_reboot_gce(
+        self, class_client: IntegrationInstance
+    ):
+        client = class_client
+        assert client.execute("test -f /run/cloud-init/cloud-id-gce").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+        client.restart()
+        assert client.execute("test -f /run/cloud-init/cloud-id-gce").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+
+    @pytest.mark.ec2
+    def test_instance_cloud_id_across_reboot_ec2(
+        self, class_client: IntegrationInstance
+    ):
+        client = class_client
+        assert client.execute("test -f /run/cloud-init/cloud-id-aws").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+        client.restart()
+        assert client.execute("test -f /run/cloud-init/cloud-id-aws").ok
+        assert client.execute("test -f /run/cloud-init/cloud-id").ok
+
```

Comment:
Let's reduce duplication in the integration tests
```suggestion
    @pytest.mark.azure
    @pytest.mark.ec2
    @pytest.mark.lxd_container
    @pytest.mark.gce
    def test_instance_cloud_id_across_reboot(
        self, class_client: IntegrationInstance
    ):
        client = class_client
        platform = client.settings.PLATFORM
        cloud_id_alias = {
            "ec2": "aws",
            "lxd_container": "lxd",
            "lxd_vm": "lxd",
        }
        cloud_file = f"cloud-id-{cloud_id_alias.get(platform, platform)}"
        assert client.execute(f"test -f /run/cloud-init/{cloud_file}").ok
        assert client.execute("test -f /run/cloud-init/cloud-id").ok
        client.restart()
        assert client.execute(f"test -f /run/cloud-init/{cloud_file}").ok
        assert client.execute("test -f /run/cloud-init/cloud-id").ok

```